src/xcode/ENA/ENA/Source/Scenes/DataDonation/DataDonationViewModels/BaseDataDonationViewModel.swift

```diff
@@ -0,0 +1,104 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import UIKit
+import OpenCombine
+
+/*** this protocol gets used to provide different view model classes for the dataDonationViewController */
+
+protocol DataDonationViewModelProtocol {
+
+	/// use these 3 to definde @published in a prorocol
+	var reloadTableView: Bool { get }
```

Comment:
I needed some time to understand what's happening here.
I think a simple closure would be sufficient enough and a lot easier to understand.

Also a data driven approach with combine, could be dynamicTableViewModel published. Every time it changes, the tableView should be reloaded.

Reply:
we use the modal as a publisher right now. This bool is obsolete, I removed it.